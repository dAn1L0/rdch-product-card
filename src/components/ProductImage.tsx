import React, { CSSProperties, useContext } from 'react'
import { ProductContext } from './ProductCard'

import noImage from '../assets/no-image.jpg'
import styles from '../styles/styles.module.css'

export interface Props {
  img?: string
  className?: string
  style?: CSSProperties
}

export const ProductImage = ({img, className, style}: Props) => {

  const { product } = useContext(ProductContext)

  let imgToShow: string
  let titleToShow: string

  if (img) {
    imgToShow = img
  } else if ( product.img ) {
    imgToShow = product.img
  } else {
    imgToShow = noImage
  }

  if ( product?.title ) {
    titleToShow = product?.title.toLowerCase().replace(' ','-')
  } else {
    titleToShow = 'product-image'
  }

  return (
    <img 
      className={ `${styles.productImg} ${className}` } 
      src={ imgToShow } 
      alt={ titleToShow }
      style={style}
    />
  )
}