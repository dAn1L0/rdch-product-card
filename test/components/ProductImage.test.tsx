import * as React from 'react';
import renderer from 'react-test-renderer'
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('Pruebas en el <ProductImage/>', () => {

  test('debe mostrar el componente correctamente con el img personalizado', () => { 
    const wrapper = renderer.create(
      <ProductImage img='https://prueba.jpg' />
    )

    expect( wrapper.toJSON() ).toMatchSnapshot();
    
  })

  test('debe mostrar el componente con el "img" del producto', () => {
    const wrapper = renderer.create(
      <ProductCard
        product={product2}
      >
        {
          () => (
            <ProductImage />
          )
        }
      </ProductCard>
    )

    expect(wrapper.toJSON()).toMatchSnapshot();
  })

})