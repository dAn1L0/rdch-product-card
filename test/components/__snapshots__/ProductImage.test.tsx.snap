// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <ProductImage/> debe mostrar el componente con el "img" del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="coffee-black"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`Pruebas en el <ProductImage/> debe mostrar el componente correctamente con el img personalizado 1`] = `
<img
  alt="product-image"
  className="productImg undefined"
  src="https://prueba.jpg"
/>
`;
